app/src/main/java/org/mozilla/focus/utils/Settings.kt

```diff
@@ -195,11 +195,18 @@ class Settings(
         preferences.edit()
             .putString(getPreferenceKey(R.string.pref_key_autoplay), prefKey)
             .apply()
-        context.components.sessionUseCases.reload.invoke(context.components.store.state.selectedTabId)
-        currentAutoplayOption = getValueByPrefKey(autoplayPrefKey = prefKey, context = context)
+        autoplayState = autoplayState.copy(
+            option = getValueByPrefKey(autoplayPrefKey = prefKey, context = context),
+            reloadSession = true
+        )
```

Comment:
This is an interesting approach but seems like a long departure from our arch direction of updating the UI and passing state through the MVI stores.
Wondering if we could rather have this state best uplifted in BrowserState.